#184752082 Setup Swagger Documentation

nishimweprince opened the pull request:
## What does this PR do?

This PR sets up Swagger Documentation for Techsmiths e-commerce-bn

## Description of Task to be completed?

- Setup Swagger Documentation using Swagger UI and Swagger JsDoc.
- Config the Swagger Docs to use the route: `/api/docs`.

## How should this be manually tested?

Clone the repo using `git clone repo_url`

- run `npm i` to install node_modules
- create a `.env` file in the root of the app and add all necessary environment variables.
   check `.env.example` to find required environment variables need to run the application.

- after starting the application, browse `http://localhost:{PORT}/api/docs` to access the documentation.

## Any background context you want to provide?

All sequelize files are excluded in `eslint` checks to avoid conflicts. Sequelize uses CommonJS which is not supported in the configuration we used.

## What are the relevant pivotal tracker/Trello stories?

- [#184752082](https://www.pivotaltracker.com/story/show/184752082)

## Screenshots (if appropriate)
<img width="1440" alt="swagger" src="https://user-images.githubusercontent.com/31816352/227601612-8bc8419c-c034-4af5-872d-bc3957d0621d.png">

jcharles921 reviewed (approved):
Excellent work prince 👌 ✨✨

Gabriel-mdv reviewed (approved):
This work is really awesome! Everything works as expected.

niyontwali reviewed (changes requested):
Please squash your commits to only 1 commit

nishimweprince committed:
New consolidated commit message

nishimweprince committed:
chore(setup) swagger documentation

- setup swagger-ui using jsdoc to write documentation using javascript
- documtations located in /src/docs which also serves as the route for the api
- add --experimental-specifier-resolution to allow node recognize imports that do not specify file extensions

Delivers #184752082
